os/tools/mkbootparam.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python
+############################################################################
+#
+# Copyright 2021 Samsung Electronics All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+#
+############################################################################
+import os
+import sys
+import struct
+
+config_path = os.path.dirname(__file__) + '/../.config'
+output_path = os.path.dirname(__file__) + '/../../build/output/bin/'
+file_path = output_path + 'bootparam.bin'
+
+FLASH_START_ADDR = 0x8000000
+
+############################################################################
+#
+# This script generates Boot Parameters for kernel.
+#
+# Boot Parameter information :
+# +------------------------------------------------+
+# | Active Idx |  First Address  | Second Address  |
+# |  (1 byte)  |    (4 bytes)    |    (4 bytes)    |
+# +------------------------------------------------+
+#  * This data is written to the last of partition with size 8Kbytes.
+#
+############################################################################
+
+def get_config_value(file_name, config):
+    with open(file_name, 'r+') as f:
+        lines = f.readlines()
+	found = False
+	for line in lines:
+		if config in line:
+			value = (line.split("=")[1])
+                        value = value.replace('"','').replace('\n','')
+			break
+	return value;
+
+def make_bootparam():
+    print "========== Making boot param start =========="
+    SIZE_OF_INDEX = 1
+    SIZE_OF_KERNEL_FIRST_ADDR = 4
+    SIZE_OF_KERNEL_SECOND_ADDR = 4
+
+    bootparam_size = SIZE_OF_INDEX + SIZE_OF_KERNEL_FIRST_ADDR + SIZE_OF_KERNEL_SECOND_ADDR
+
+    names = get_config_value(config_path, "CONFIG_FLASH_PART_NAME=").split(",")
+    sizes = get_config_value(config_path, "CONFIG_FLASH_PART_SIZE=").split(",")
+    names = filter(None, names)
+    sizes = filter(None, sizes)
```

Comment:
Is this filter necessary?

Reply:
It is for removing empty element from a list. The empty string is caused by `,` at the end of config value.

Reply:
1. If we don't remove empty element, what is the problem?
2. Can we guarantee or give guidance that if name is none, size should be none as well?
   If we don't size and name could be mismatched, then it will cause a problem.

Reply:
1. An error occurs about null element in line 76.
2. I added verifying size about what you said in line 88.